src/Emitter.ts

```diff
@@ -19,91 +19,298 @@ export enum Events {
+   * const handler = ( ) => { console.log('baz') }
+   *
+   * // subscribe to the 'namedExpressionAdded' event, pass the handler
+   * hfInstance.on('namedExpressionAdded', handler);
+   *
+   * // add a named expression
+   * // the console prints 'baz' each time a named expression is added
+   * const changes = hfInstance.addNamedExpression('prettyName', '=Sheet1!$A$1+100', 0);
+   *
+   * // unsubscribe from the 'namedExpressionAdded' event
+   * hfInstance.off('namedExpressionAdded', handler);
+   *
+   * // add another named expression
+   * // this time, the console doesn't print anything
+   * const changes = hfInstance.addNamedExpression('uglyName', '=Sheet1!$A$1+100', 0);
+   * ```
+   * 
    * @category Named Expression
    */
   namedExpressionAdded: (namedExpressionName: string, changes: ExportedChange[]) => any,
 
   /**
-   * The event is triggered upon removing a named expression with specified values and from an indicated location.
+   * Occurs when a named expression with specified values is removed and from an indicated location.
    * 
    * @event
    * 
    * @param {string} namedExpressionName the name of removed expression
    * @param {ExportedChange[]} changes the values and location of applied changes
+   * 
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray([
+   * ['42'],
+   * ]);
+   *
+   * // define a function to be called when the event occurs
+   * const handler = ( ) => { console.log('baz') }
+   * 
+   * // subscribe to the 'namedExpressionRemoved' event, pass the handler
+   * hfInstance.on('namedExpressionRemoved', handler);
+   *
+   * // add some named expressions
+   * hfInstance.addNamedExpression('prettyName', '=Sheet1!$A$1+100', 0);
+   * hfInstance.addNamedExpression('uglyName', '=Sheet1!$A$1+100', 0);
+   * 
+   * // remove a named expression
+   * // the console prints 'baz' each time a named expression is removed
+   * const changes = hfInstance.removeNamedExpression('prettyName', 0);
+   *
+   * // unsubscribe from the 'namedExpressionRemoved' event
+   * hfInstance.off('namedExpressionRemoved', handler);
    *
+   * // remove another named expression
+   * // this time, the console doesn't print anything
+   * const changes = hfInstance.removeNamedExpression('uglyName', 0);
+   * ```
+   * 
    * @category Named Expression
    */
   namedExpressionRemoved: (namedExpressionName: string, changes: ExportedChange[]) => any,
 
   /**
-   * The event is triggered upon changing values in a specified location.
+   * Occurs when values in a specified location are changed and cause recalculation.
    * 
    * @event
    * 
    * @param {ExportedChange[]} changes the values and location of applied changes
    *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildFromArray([
+   * ['1', '2', '=A1'],
+   * ]);
+   *
+   * // define a function to be called when the event occurs
+   * const handler = ( ) => { console.log('baz') }
```

Comment:
We've got changes here as well

```suggestion
   * const handler = ( changes ) => { console.log('baz') }
```